src/pydocstyle/checker.py

```diff
@@ -43,36 +44,40 @@ class PEP257Checker(object):
 
     """
 
-    def check_source(self, source, filename):
+    def check_source(self, source, filename, ignore_decorators):
         module = parse(StringIO(source), filename)
         for definition in module:
-            for check in self.checks:
+            for this_check in self.checks:
                 terminate = False
-                if isinstance(definition, check._check_for):
-                    if definition.skipped_error_codes != 'all':
-                        error = check(None, definition, definition.docstring)
+                if isinstance(definition, this_check._check_for):
+                    if definition.skipped_error_codes != 'all' and \
+                            not any(ignore_decorators is not None and
+                                    len(ignore_decorators.findall(dec.name))
+                                    for dec in definition.decorators):
```

Comment:
This `if` is too long. Break it up by assigning to local variables, something like:
```.py
checking_for_errors = ..
skipping_decorator = ..
if checking_for_errors and not skipping_decorator:
    # run the check
```
Think about better names, though :)